birdhouse/components/weaver/post-docker-compose-up

```diff
@@ -0,0 +1,149 @@
+#
+# This script is executed after 'docker-compose up' of the full bird-house stack.
+# The reason why WPS providers must be registered after starting everything is because Weaver attempts to query
+# them (with GetCapabilities requests) to obtain their metadata.
+# Because some WPS containers could take longer to start, it is otherwise not guaranteed that they would be ready
+# to receive those requests before Weaver starts querying them.
+#
+# Parameters:
+#
+#   WEAVER_WPS_PROVIDERS:
+#       list of provider names (comma or space delimited), all are assumed to be available at
+#       "https://${PAVICS_FQDN_PUBLIC}${TWITCHER_PROTECTED_PATH}/<provider-name>"
+#   WEAVER_WPS_PROVIDERS_MAX_TIME:
+#       limit script execution up to maximum this number of seconds
+#
+# Following configurations are expected to be inherited from bird-house/weaver-component env.local/default.env:
+#   - MAGPIE_ADMIN_USERNAME
+#   - MAGPIE_ADMIN_PASSWORD
+#   - PAVICS_FQDN_PUBLIC
+#   - TWITCHER_PROTECTED_PATH
+#   - WEAVER_MANAGER_NAME
+#
+
+MAGPIE_URL="https://${PAVICS_FQDN_PUBLIC}/magpie"
+WEAVER_URL="https://${PAVICS_FQDN_PUBLIC}${TWITCHER_PROTECTED_PATH}/${WEAVER_MANAGER_NAME}"
+WEAVER_WPS_PROVIDERS_MAX_TIME=${WEAVER_WPS_PROVIDERS_MAX_TIME:-60}
+
+# logging
+YELLOW=${YELLOW:-$(tput setaf 3)}
+RED=${RED:-$(tput setaf 1)}
+NORMAL=${NORMAL:-$(tput sgr0)}
+PREFIX="[Weaver] "
+ERROR="${PREFIX}${RED}ERROR${NORMAL}: "
+WARNING="${PREFIX}${YELLOW}WARNING${NORMAL}: "
+
+if [ -z "${WEAVER_WPS_PROVIDERS}" ]; then
+    echo "${WARNING}Nothing specified in WEAVER_WPS_PROVIDERS to register WPS remote providers."
+    exit 0
+fi
+
+# registration of WPS providers require authenticated access, obtain login from Magpie
+start_time="$(date -u +%s)"
+while true; do
+    # login (output null + cookie-jar '-' redirects output cookie to variable)
+    cookie_jar=$( \
+        curl -k -s -m 2 -o /dev/null \
+             -X POST \
+             -H "Content-Type: application/json" \
+             -d "{\"user_name\": \"${MAGPIE_ADMIN_USERNAME}\", \"password\": \"${MAGPIE_ADMIN_PASSWORD}\"}" \
+             --cookie-jar - \
+             "${MAGPIE_URL}/signin" \
+    )
+    # trim excess stuff in cookie_jar pseudo-file (comments, empty lines)
+    cookie_jar=$(echo "${cookie_jar}" | grep -v '# ' | grep -v -e '^$')
+    # validate exactly 1 cookie retrieved (empty if bad-auth or invalid endpoint)
+    if [ "$(echo "${cookie_jar}" | wc -l)" -eq 1 ]; then
+        fields=$(echo "${cookie_jar}" | wc -w)
+        cookie_name=$(echo "${cookie_jar}" | cut -f $(( fields - 1 )) )
+        cookie_value=$(echo "${cookie_jar}" | cut -f "${fields}")
+        cookie="${cookie_name}=${cookie_value}"
+        break;
+    fi
+    # interrupt if max time reached
+    next_time=$(date -u +%s)
+    delta_time=$(( next_time - start_time ))
+    if [ ${delta_time} -ge "${WEAVER_WPS_PROVIDERS_MAX_TIME}" ]; then
+        msg="Failed to register all provider specified in WEAVER_WPS_PROVIDERS."
```

Comment:
Copy/paste error?  This is magpie login step, not provider registering step.

Reply:
I can add another extra message on top to highlight the step (I was using only the exit code for that).
I wanted to inform more about the whole procedure failing.